doc/spec.md

```diff
@@ -3755,6 +3768,68 @@ y = set([2, 3])
 x.union(y)                              # set([1, 2, 3])
 ```
 
+<a id='set·add'></a>
+### set·add
+
+`S.add(x)` adds the value `x` to the set `S`. This has no effect if `x` is already in `S`.
+
+```python
+x = set([1, 2])
+x.add(3)                             # None (x == set([1, 2, 3]))
+x.add(3)                             # None (x == set([1, 2, 3]))
+```
+
+<a id='set·pop'></a>
+### set·pop
+
+`S.pop()` removes an arbitrary item from the set and returns it.
+
+`pop` fails if the set is empty.
+
+```python
+x = set([1, 2, 3])
+x.pop()                              # 1 (x == set([2, 3]))
+```
+
+<a id='set·remove'></a>
+### set·remove
+
+`S.remove(x)` removes `x` from the set and returns None.
+
+`remove` fails if the set does not contain `x` or is frozen.
```

Comment:
All the mutating methods should ideally mention what happens when the set is frozen, especially given the subtlety that `add` may not always attempt to mutate the set (if the element is already present), so it does not always fail on a frozen dict. The spec should probably take a position on this, by stating that the lookup precedes the attempt at mutation (that's what dict.setdefault does, although the document is not particularly explicit).

Reply:
Good point, all of the frozen checks defer to the underlying hashtable so I wasn't thinking about this case. I'll verify and document this behaviour. 

Related - how do you feel about calling `clear()` on an empty, frozen dict or set? Should that succeed?

Reply:
Landed on the following semantics around frozen sets:
 - `set.clear()` will succeed on an empty frozen set
 - `set.discard(x)` will succeed on a frozen set which does not contain `x`
 - `set.add(x)` will succeed on a frozen set which already contains `x`
 - `pop()` and `remove()` will always fail on frozen sets with `cannot delete from frozen hash table`. 
 
 Two edge cases to consider - 
  - Should `pop()` on an empty frozen set fail because its empty, or because its frozen?
  - Should `remove(x)` on a frozen set not containing x fail because of missing key, or because its frozen?

Reply:
> Should pop() on an empty frozen set fail because its empty, or because its frozen?
> Should remove(x) on a frozen set not containing x fail because of missing key, or because its frozen?

Generally we don't specify which error you get when more than one applies. What's more important is that we specify whether successful shortcuts trump slow-path errors (as in the case of discarding an unhashable from an empty set: does it fail because unhashable or succeed because empty?). So what you have looks fine.